contract/models/contract.py

```diff
@@ -11,19 +11,36 @@
 from odoo.tools.translate import _
 
 
-class AccountAnalyticAccount(models.Model):
-    _name = 'account.analytic.account'
-    _inherit = [
-        'account.analytic.account',
-        'account.abstract.analytic.contract',
-    ]
+class ContractContract(models.Model):
+    _name = 'contract.contract'
+    _inherit = ['mail.thread', 'contract.abstract.contract']
 
+    active = fields.Boolean(
+        default=True,
+    )
+    code = fields.Char(
+        string="Reference",
+    )
+    group_id = fields.Many2one(
+        string="Group",
+        comodel_name='account.analytic.account',
+        ondelete='restrict',
+    )
+    analytic_account_id = fields.Many2one(
+        string="Analytic account",
+        comodel_name='account.analytic.account',
+    )
+    currency_id = fields.Many2one(
+        related="company_id.currency_id",
+        string="Currency",
+        readonly=True,
+    )
     contract_template_id = fields.Many2one(
-        string='Contract Template', comodel_name='account.analytic.contract'
+        string='Contract Template', comodel_name='contract.template'
     )
     recurring_invoice_line_ids = fields.One2many(
```

Comment:
what do you think if we rename this field to contract_line_ids?

Reply:
Good idea, this name is confusing.